source/material_model/steinberger.cc

```diff
@@ -362,51 +334,46 @@ namespace aspect
           max_lateral_eta_variation    = prm.get_double ("Maximum lateral viscosity variation");
           thermal_conductivity_value = prm.get_double ("Thermal conductivity");
 
-
           // Parse the table lookup parameters
-
-          // The Steinberger material model currently assumes that all the
-          // compositional fields correspond to materials with
-          // PerpleX lookup tables.
-          // Therefore the first composition index is hard-coded as zero and
-          // a background field exists if the number of files is one greater
-          // then the number of compositional fields.
-          prm.set("Material file format", "perplex");
-          prm.set("Index of first mass fraction compositional field", "0");
-          if (material_file_names.size() == this->n_compositional_fields() + 1)
-            prm.set("Background material", "true");
-          else
-            prm.set("Background material", "false");
           equation_of_state.initialize_simulator (this->get_simulator());
           equation_of_state.parse_parameters(prm);
 
+          // Assign background field and do some error checking
+          if (equation_of_state.number_of_lookups() == this->n_compositional_fields() + 1)
+            {
+              prm.set("Background material", "true");
+            }
+          else
+            {
+              AssertThrow ((equation_of_state.number_of_lookups() == this->n_compositional_fields()),
+                           ExcMessage("The Steinberger material model assumes that all compositional "
+                                      "fields correspond to mass fractions of materials. There must either be "
+                                      "the same number of material lookup files as compositional fields, "
+                                      "or one additional file (if a background field is used)."));
+
+              prm.set("Background material", "false");
+            }
+
+          AssertThrow (prm.get_integer ("Index of first mass fraction compositional field") == 0,
+                       ExcMessage("The Steinberger material model currently assumes that all the "
+                                  "compositional fields correspond to materials with PerpleX lookup tables. "
+                                  "Therefore the 'Index of first mass fraction compositional field' "
+                                  "parameter must be equal to zero. "));
+
           prm.leave_subsection();
         }
         prm.leave_subsection();
 
-        // Do some error checking
-        AssertThrow ((material_file_names.size() == 1) ||
-                     (material_file_names.size() == this->n_compositional_fields()) ||
-                     (material_file_names.size() == this->n_compositional_fields() + 1),
-                     ExcMessage("This material model expects either one material data file, or as many files as compositional fields, "
```

Comment:
> Now having only one data file is not allowed any more
Sure it is! It's allowed if the number of compositional fields is zero (background field) or one (no background field).

Reply:
Yes, but before it was allowed for any number of compositional fields. Basically, if you only have one table, it's clear that you want to use that, so you can use as many compositional fields as you want for whatever else you want. That now doesn't work any more. So for backwards compatibility, I would change the Assert above so that it also allows for having one table independently of the number of compositional fields.

Reply:
Oh, holy squirrels, you're right. Sorry, I had misunderstood/forgotten that a single file could be used for multiple material fields. I've relaxed the assert.